golang/internal/runtime/container/container.go

```diff
@@ -0,0 +1,120 @@
+package containerruntime
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/docker/client"
+	"github.com/hashicorp/go-version"
+	"github.com/rs/zerolog/log"
+)
+
+// If a runtime version doesn't satisfy recommendation: show warning
+// If a runtime version doesn't satisfy minimum: failure\
+
+const (
+	Podman                         = "podman"
+	Docker                         = "docker"
+	MinimumDockerServerVersion     = "20.10.0"
+	RecommendedDockerServerVersion = "23.0.0"
+	MinimumPodmanServerVersion     = "4.0.0"
+	RecommendedPodmanServerVersion = "4.4.0"
+	PodmanHost                     = "host.containers.internal"
+	DockerHost                     = "host.docker.internal"
+)
+
+func VersionCheck() {
```

Comment:
ctx and client may be pluggable, for testing and different use cases, where we already have a client or have a valid context already.

Reply:
Also the input should be `serverVersion.Version` for testability.

Reply:
Refactored a bit, thanks!